docs/content/testing/writing-a-new-test.md

```diff
@@ -0,0 +1,314 @@
+* SptialOS Unity GDK documentation - [Testing guidelines](./test-guidelines.md)
+* NUnit video - [Introduction to NUnit](https://www.youtube.com/watch?v=1TPZetHaZ-A)<br/>
+**Skip** the part between 0:40 - 2:49 as it does not apply to Unity testing. <br/>
+**Skip** the part between 5:25 - 7:57 as it explains how to use the NUnit Runner and we’ll use Unity Test Runner instead.
+* NUnit video - [Unit Testing Using NUnit](https://www.codeproject.com/articles/178635/unit-testing-using-nunit)
+- Unity documentation - [Writing and executing tests in Unity Test Runner](https://docs.unity3d.com/Manual/PlaymodeTestFramework.html)
+* Infallible Code video -  [How To Test Unity ECS Code](https://http://infalliblecode.com/test-unity-ecs/) (Note the code for this is behind a paywall.)
+
+## Example new test
+
+The example below uses a made-up class `MyClass`.
+
+Let's say you have this class:
+
+```cs
+using System;
+
+namespace Improbable.Gdk.Core.Utility
+{
+    public class MyClass : IDisposable
+    {
+        public int InstanceValue;
+
+        public MyClass()
+        {
+                // Pretend to set up some state.
+                InstanceValue = 0;
+        }
+
+        public void DoInstanceOperation()
+        {
+                InstanceValue++;
+        }
+
+        public void DoSomethingElse()
+        {
+                InstanceValue += 5;
+        }
+
+        public static int AddTwoNumbers(int a, int b)
+        {
+                return a + b;
+        }
+
+        public void Dispose()
+        {
+                // Pretend to clean up some state.
+        }
+    }
+}
+```
+
+This class would exist at this path: `workers/Unity/Packages/com.improbable.gdk.core/Utility/MyClass.cs`.
+
+### Test folder and filename
+
+The package for this class would be: `workers/Unity/Packages/com.improbable.gdk.core/`.
+
+The path within the package would be: `Utility/MyClass.cs`.
+
+Following the [package directory structure instructions](./testing-guidelines.md#package-directory-structure), the test filename would be:
+
+`workers/Unity/Assets/Gdk/Core/Tests/EditMode/Utility/MyClassTests.cs`
+
+(This assumes that the `workers/Unity/Packages/com.improbable.gdk.core/Tests/` directory exists.
+ If it does not exist, see the section on 
+ [How to  create a new test folder and assembly](#how-to-create-a-new-test-folder-and-assembly).).
+
+### Test fixtures
+
+With NUnit, you can add a `TestFixture` attribute to a class. This makes the class a test. While you create specific tests for specific methods in specific classes, you can reuse the tests where methods have similar setups. (A class with a `TestFixture` attribute is also known as “test fixture” and “fixture”.) See NUnit’s documentation on the [TestFixture attribute](https://github.com/nunit/docs/wiki/TestFixture-Attribu te) and the [test fixture example class](https://github.com/nunit/docs/wiki/TestFixture-Attribute#example-1) in the the NUnit repository.
```

Comment:
broken link, should be https://github.com/nunit/docs/wiki/TestFixture-Attribute

Reply:
Not broken for me, it should jump to the right example